src/isc-dhcp/patch/0013-Add-new-option-si-to-use-source-interface-IP-while-r.patch

```diff
@@ -0,0 +1,174 @@
+From 711bcb53f28603c4b0a2d18b6f4354b508d68b35 Mon Sep 17 00:00:00 2001
+From: Joe LeVeque <jolevequ@microsoft.com>
+Date: Wed, 4 Mar 2020 19:46:21 +0000
+Subject: [PATCH] Add new option -si to use source interface IP while relaying
+ packets to upstream ports
+
+---
+ common/socket.c  | 96 ++++++++++++++++++++++++++++++++++++++++++++++++
+ includes/dhcpd.h |  1 +
+ relay/dhcrelay.c |  8 ++++
+ 3 files changed, 105 insertions(+)
+
+diff --git a/common/socket.c b/common/socket.c
+index e8851b4..4e88945 100644
+--- a/common/socket.c
++++ b/common/socket.c
+@@ -78,6 +78,29 @@ static unsigned int global_v4_socket_references = 0;
+ static int global_v4_socket = -1;
+ #endif
+ 
++/*
++ * If set, uses "from" interface IP for packet Tx.
++ * If not set, kernel chooses appropriate src ip for tx pkts
++ */
++int use_src_intf_ip_for_tx;
++
++/*
++ * For both send_packet6() and receive_packet6() we need to allocate
++ * space for the cmsg header information.  We do this once and reuse
++ * the buffer.  We also need the control buf for send_packet() and
++ * receive_packet() when we use a single socket and IP_PKTINFO to
++ * send the packet out the correct interface.
++ */
++static void   *v4_control_buf = NULL;
++static size_t  v4_control_buf_len = 0;
++
++static void
++v4_allocate_cmsg_cbuf(void) {
++	v4_control_buf_len = CMSG_SPACE(sizeof(struct in_pktinfo));
++	v4_control_buf = dmalloc(v4_control_buf_len, MDL);
++	return;
++}
++
+ /*
+  * If we can't bind() to a specific interface, then we can only have
+  * a single socket. This variable insures that we don't try to listen
+@@ -629,6 +652,78 @@ ssize_t send_packet (interface, packet, raw, len, from, to, hto)
+ 	struct hardware *hto;
+ {
+ 	int result;
++#if 1
```

Comment:
why do we use #if 1 here, suggest to remove.

Reply:
Removed in latest commit.